### PR TITLE
Ytdraft

### PR DESCRIPTION
**Pull request by yuntianhu**
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate action.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code are 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added, or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/InvoiceHelper.java [hunk 1]
```diff
@@ -349,6 +351,10 @@ private Future<Void> validateAndHandleInvoiceStatusTransition(Invoice invoice, I
         validator.validateInvoice(invoice, invoiceFromStorage);
         verifyUserHasManagePermission(invoice.getAcqUnitIds(), invoiceFromStorage.getAcqUnitIds(), okapiHeaders);
         verifyUserHasFiscalYearUpdatePermission(invoice.getFiscalYearId(), invoiceFromStorage.getFiscalYearId(), okapiHeaders);
+        //verify if the user can approve the inovice 
+       verifyApprovalPermission(invoice.getAcqUnitIds(), invoiceFromStorage.getAcqUnitIds(), okapiHeaders);
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/rest/impl/InvoiceHelper.java:355]: it is not necessary to provide acq units here

File: src/main/java/org/folio/rest/impl/InvoiceHelper.java [hunk 2]
```diff
@@ -349,6 +351,10 @@ private Future<Void> validateAndHandleInvoiceStatusTransition(Invoice invoice, I
         validator.validateInvoice(invoice, invoiceFromStorage);
         verifyUserHasManagePermission(invoice.getAcqUnitIds(), invoiceFromStorage.getAcqUnitIds(), okapiHeaders);
         verifyUserHasFiscalYearUpdatePermission(invoice.getFiscalYearId(), invoiceFromStorage.getFiscalYearId(), okapiHeaders);
+        //verify if the user can approve the inovice
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/rest/impl/InvoiceHelper.java:354]: better to rename method and after it code will be self describing and we will not need comment any more,
you can use this nameL verifyUserHasInvoiceApprovePermission.
The same we should do with other method

File: src/main/java/org/folio/rest/impl/InvoiceHelper.java [hunk 3]
```diff
@@ -25,7 +25,9 @@
 import static org.folio.utils.UserPermissionsUtil.verifyUserHasAssignPermission;
 import static org.folio.utils.UserPermissionsUtil.verifyUserHasManagePermission;
 import static org.folio.utils.UserPermissionsUtil.verifyUserHasFiscalYearUpdatePermission;
-
+import static org.folio.utils.UserPermissionsUtil.verifyUserHasFiscalYearUpdat;
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/rest/impl/InvoiceHelper.java:28]: this code will not compile, check that code is compling before commit

File: src/main/java/org/folio/utils/UserPermissionsUtil.java [hunk 4]
```diff
@@ -63,6 +63,42 @@ public static void verifyUserHasManagePermission(List<String> newAcqUnitIds, Lis
     }
   }
 
+  public static void verifyPaidPermission(List<String> newAcqUnitIds, List<String> currentAcqUnitIds, String acqSataus,
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/utils/UserPermissionsUtil.java:66]: what is means acqStatus? we can rename to invoiceStatus

File: src/main/java/org/folio/rest/impl/InvoiceHelper.java (hunk 3 above)
Comment (yuntianhu) [reply src/main/java/org/folio/rest/impl/InvoiceHelper.java:28]: sure, I am sorry.

File: src/main/java/org/folio/utils/UserPermissionsUtil.java (hunk 4 above)
Comment (yuntianhu) [reply src/main/java/org/folio/utils/UserPermissionsUtil.java:66]: sure. will do

File: src/main/java/org/folio/utils/UserPermissionsUtil.java [hunk 5]
```diff
@@ -84,4 +120,16 @@ private static boolean isManagePermissionRequired(Set<String> newAcqUnits, Set<S
...
+    /**
+   * The method checks whether the user has the desired permission to update the fiscal year.
+   *
+   * @throws HttpException if user does not have fiscal year update permission
+   * @param newFiscalYearId     fiscal year id coming from request
+   * @param fiscalYearIdFromStorage fiscal year id from storage
+   */
+
+  private static boolean isApporved(String newStatus) {
+    return !CollectionUtils.isEqualCollection(newStatus, "Approved");
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/utils/UserPermissionsUtil.java:132]: if you want to compare strings you should not use collections, you can use equals method or StringUtils

File: src/main/java/org/folio/rest/impl/InvoiceHelper.java [hunk 6]
```diff
@@ -349,6 +351,10 @@ private Future<Void> validateAndHandleInvoiceStatusTransition(Invoice invoice, I
         validator.validateInvoice(invoice, invoiceFromStorage);
         verifyUserHasManagePermission(invoice.getAcqUnitIds(), invoiceFromStorage.getAcqUnitIds(), okapiHeaders);
         verifyUserHasFiscalYearUpdatePermission(invoice.getFiscalYearId(), invoiceFromStorage.getFiscalYearId(), okapiHeaders);
+        //verify if the user can approve the inovice 
+       verifyApprovalPermission(invoice.getAcqUnitIds(), invoiceFromStorage.getAcqUnitIds(), okapiHeaders);
+        //verify if the invoice is allowed to be paid 
+       verifyPaidPermission(invoice.getAcqUnitIds(), invoiceFromStorage.getAcqUnitIds(), invoice.getStatus().toString(), okapiHeaders);
```
[the remark this answers is not in the document]
Comment (SerhiiNosko) [reply src/main/java/org/folio/rest/impl/InvoiceHelper.java:357]: Also you think to provide 2 statuses, from requested Invoice and from Invoice from DB and do it for both methods.
And in these methods you should check that statuses not equals, and new status(requested) equals Paid or Approved

[the remark this answers is not in the document]
Comment (yuntianhu) [reply src/main/java/org/folio/rest/impl/InvoiceHelper.java:357]: ok,  will do.

**Pull request closed without merging**